internal/charmstore/addentity.go

```diff
@@ -437,7 +437,7 @@ func (s *Store) addCharm(c charm.Charm, p addParams) (err error) {
 	// the name of a bundle. This is racy, but it's the best we can
 	// do. Also check that there isn't an existing multi-series charm
 	// that would be replaced by this one.
-	entities, err := s.FindEntities(entity.BaseURL, nil)
+	entities, err := s.FindEntities(entity.BaseURL, mongodoc.UnpublishedChannel, nil)
```

Comment:
Every entity, by definition is in the unpublished channel, perhaps we need to change the name of the unpublished channel to make the meaning more clear.